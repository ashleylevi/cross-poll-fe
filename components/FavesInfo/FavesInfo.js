import React from 'react';
import { StyleSheet, Text, View, Button, TouchableOpacity, ImageBackground, Image, ScrollView } from 'react-native';
import { Icon } from 'react-native-elements';
import Loading from '../Loading/Loading';
import { cleanPet } from '../helpers/helpers';

export default class FavesInfo extends React.Component {
  constructor(props) {
    super(props)
  }

  render() {
    const { currentPet, shelter } = this.props;
    const image = currentPet.photos[2]

    if (currentPet) {
      return (
        <View>
          <ImageBackground source = {{uri: image}} style={styles.moreInfoImage}
          imageStyle={styles.borderRad}>
            <Text style={styles.petName}>{currentPet.name} </Text>
            <View style={styles.currentPetBreedAge}>
              <Text style={{color:'white'}}>{currentPet.breed}</Text>
              <Text style={{color:'white'}}>{currentPet.age}</Text>
            </View>
          </ImageBackground>
            <ScrollView style={styles.scroll}>
              <Text style={{fontSize: 15}}>{currentPet.description}</Text>
              <Text style={{fontSize: 15}}>{currentPet.name} can be found at </Text>
              <View style={styles.shelterInfo}>
              <Text style={styles.phone}>{shelter.phone}</Text>
              <Text style={styles.cityStateZip}>{shelter.city}{shelter.state}{shelter.zip}</Text>
            </View>
            <TouchableOpacity
            style={styles.contactButton}>
              <Text style={styles.contactButtonText}> Contact {shelter.name} </Text>
            </TouchableOpacity>
            <TouchableOpacity
            style={styles.backButton}> 
            </TouchableOpacity>
            </ScrollView>
        </View>
      )
    } else {
      return (
        <Loading  />
      )
    }
    
    
  }
}

const styles = StyleSheet.create({
  petName: {
    fontSize: 40,
    color: 'white',
    position: 'absolute',
    bottom: 60,
    left: 20
  },
  swiper: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    width: 500
  },
  image: {
    height: 500,
    width: 350,
  },
  moreInfoImage: {
    marginTop: 5
  },
  borderRad: {
    borderColor: 'white',
    borderRadius: 30,
    borderWidth: 2
  },
  contactButton: {
    backgroundColor: 'white',
    borderColor: 'black',
    borderRadius: 24,
    borderStyle: 'solid',
    borderWidth: 2,
    fontSize: 20,
    height: 30,
    shadowColor: '#000',
    shadowOffset: { width: 2, height: 4 },
    shadowOpacity: 0.8,
    shadowRadius: 4,
    textAlign: 'center',
    width: 300,
    marginBottom: 35,
    marginLeft: 23,
    marginTop: 10
  },
  contactButtonText: {
    textAlign: 'center',
    fontSize: 15,
    marginTop: 3
  },
  hamburgerIcon: {
    height: 50,
    width: 50,
    marginBottom: 40
  },
  hamburgerContainer: {
    height: 50,
    width: 50,
    position: 'relative',
    right: 120,
    top: 40,
    marginBottom: 40
  },
  description: {
    height: 200,
    width: 300,
    flex: 1,
    textAlign: 'center',
    marginTop: 10,
    fontSize: 15
  },
  canBeFound: {
    textAlign: 'center',
    marginTop: 5,
    marginBottom: 0,
    fontSize: 15
  },
  shelterName: {
    color: 'white',
    fontSize: 15,
    position: 'absolute',
    top: 415,
    left: 20,
    marginTop: 20
  },
  shelterInfo: {
    marginTop: 0,
    fontSize: 10,
    textAlign: 'center'
  },
  phone: {
    textAlign: 'center'
  },
  cityStateZip: {
    textAlign: 'center'
  },
  petBreedAge: {
    position: 'absolute',
    top: 240,
    left: 20
  },
  moreInfoImage: {
    height: 300,
    width: 350,
    marginTop: 10  
  },
  scroll: {
    width: 300,
    height: 80,
    marginLeft: 30,
    marginRight: 10
  },
  cog: {
    position: 'absolute',
    top: -10
  },
  heart: {
    position: 'absolute',
    top: -58, 
    left: 100
  }
})


