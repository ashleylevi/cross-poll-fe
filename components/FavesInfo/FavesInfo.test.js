import React from 'react'; 
import { shallow, mount } from 'enzyme';
import FavesInfo from '../FavesInfo/FavesInfo';
import renderer from 'react-test-renderer';

describe('FavesInfo', () => {
  let mockCurrentPet;
  let mockShelter;

  beforeEach(() => {
    mockCurrentPet = {
      animal: 'Cat',
      breed: 'Domestic Short Hair',contactInfo: {},
      description: '12 year old..',
      id: '42736275',
      name: 'Zoey',
      photos: ['http://photos.petfinder.com/photos/pets/42736275/1/?bust=1536778269&width=60&-pnt.jpg',
      'http://photos.petfinder.com/photos/pets/42736275/1/?bust=1536778269&width=95&-fpm.jpg',
      'http://photos.petfinder.com/photos/pets/42736275/1/?bust=1536778269&width=500&-x.jpg',
      'http://photos.petfinder.com/photos/pets/42736275/1/?bust=1536778269&width=300&-pn.jpg'],
      sex: 'F',
      shelterId: 'CO186',
      size: 'M'
    };

    mockShelter = {
      city: 'Denver',
      email: 'barnwatercat@aol.com',
      id: 'CO186',
      latitude: '39.6788',
      longitude: '-104.9636',
      name: 'Barnwater Cats Rescue Organization',
      phone: '(303) 759-2855',
      state: 'CO',
      zip: '80210'
    };
  })

  it('should match the snapshot with all data passed in correctly', () => { 

    const wrapper = shallow( <FavesInfo 
      currentPet={mockCurrentPet}
      shelter={mockShelter}
    />);

    expect(wrapper).toMatchSnapshot(); 
  })
});