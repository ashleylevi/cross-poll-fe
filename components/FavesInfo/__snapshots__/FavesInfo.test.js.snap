// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavesInfo should match the snapshot with all data passed in correctly 1`] = `ShallowWrapper {}`;
